### PR TITLE
Command interpreter, command builder and third party dependencies

### PR DESCRIPTION
**Pull request by MrRoaming**
adding the **_Command Builder_** and its _**Third Party Dependencies**_

### CODE REVIEW

File: CONVERTER_SERVICE/src/controller/image/Command.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+class Command {
+
+    constructor(id = '', value) {
+
+        this.id = id;
+        this.value = value;
+    }
+
+    //get and set the "id"
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/Command.js:9]: Erase the comment

File: CONVERTER_SERVICE/src/controller/image/Command.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    getId() {
+
+        return this.id;
+    }
+    setId(id) {
+
+        this.id = id;
+    }
+
+    //get and set the value
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/Command.js:20]: Erase the comment

File: CONVERTER_SERVICE/src/controller/image/Command.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        return this.value;
+    }
+    setValue(value) {
+
+        this.value = value;
+    }
+
+}
+
+module.exports = Command;
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/Command.js:33]: Need the last line required

File: CONVERTER_SERVICE/src/controller/image/CommandBuilder.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+        for (let property in object) {
+
+            let commandCreator = new Command(property, object[property])
+            let result = '';
+
+            result = this.interpreter.validateCommand(commandCreator) + commandCreator.getValue();
+            finalCommand += result;
+        }
+        //Only for testing purposes
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/CommandBuilder.js:24]: Erase the comment

File: CONVERTER_SERVICE/src/controller/image/CommandBuilder.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        console.log(finalCommand);
+    }
+}
+
+//Harcoded for testing purposes
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/CommandBuilder.js:29]: Erase the comment

File: CONVERTER_SERVICE/src/controller/image/CommandBuilder.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+new CommandBuilder({ grayScaleImage: true, resizeImage: 'string', qualityImage: 1 });
+
+module.exports = CommandBuilder;
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/CommandBuilder.js:32]: Last line required

File: CONVERTER_SERVICE/src/controller/image/CommandSchema.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+class CommandSchema {
+
+    constructor(id = '', command = '') {
+
+        this.id = id;
+        this.command = command;
+    }
+
+    //get and set the id
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/CommandSchema.js:9]: Erase the comment

File: CONVERTER_SERVICE/src/controller/image/CommandSchema.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    getId() {
+
+        return this.id;
+    }
+    setId(id) {
+
+        this.id = id;
+    }
+
+    //get and set the command
```
Comment (serqiiiio) [inline CONVERTER_SERVICE/src/controller/image/CommandSchema.js:20]: Erase the comment

**Review by serqiiiio — CHANGES_REQUESTED** [review]
Most of the reviews are for comments in the code, also there isn't last line required  of our protocol 

But it's really nice to read your code !!

**Review by virichino — COMMENTED** [review]
Erase commented code, and there are unnecessary spaces between code lines

**Pull request closed without merging**